test/extensions/filters/http/cache/cache_filter_integration_test.cc

```diff
@@ -76,16 +76,22 @@ class CacheIntegrationTest : public Event::TestUsingSimulatedTime,
     return response_decoder;
   }
 
-  std::function<void()>
-  simulateUpstreamResponse(const Http::TestResponseHeaderMapImpl& headers,
-                           OptRef<const std::string> body,
-                           OptRef<const Http::TestResponseTrailerMapImpl> trailers) {
+  // split_body allows us to test the behavior when encodeData is in more than one part.
```

Comment:
no new integration tests?  I think you should try for a best-effort backup test if at all possible.

Reply:
I think the existing integration tests have pretty good coverage. A new integration test wouldn't really make sense because whatever different behavior we're trying to test (compared to what's already covered) would need a special cache implementation for just that behavior, at which point we'd be mostly just integration-testing test code. At which point it might as well just be one of the unit tests that was already added.
(There is the "body in multiple parts" behavior, that was rolled into an existing test case, because that was previously not covered.)

That said, I'm not sure what is meant by a best-effort backup test, so maybe that's a good idea?

Reply:
yeah sorry by best-effort backup tests I meant a test where we likely back up.  So for downstream backup for example we have the ability to fake EAGAIN on the downstream connection so when we're sending a large response we'll force backup.  I think having a way to inject events to make the cache look slow so we can e2e test this would be pretty valuable - personally I wouldn't trust an implementation without this tested but as the cache is marked WIP I won't insist on it but I would very strongly encourage it.  having debugged similar issues you never want to in production =P

Reply:
Filed #28809 for this. Definitely would not be comfortable rolling it in as part of this change.